src/main/java/com/suchocki/parkingmeter/entity/ParkAction.java

```diff
@@ -0,0 +1,108 @@
+package com.suchocki.parkingmeter.entity;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public class ParkAction {
+	private int id; // this will be primary key in database
+	private Date start;
+	private Date end;
+	private Driver driver;
+
+	public ParkAction() {
+
+	}
+
+	public ParkAction(Date start, Driver driver) {
+		this.start = start;
+		this.driver = driver;
+	}
+
+	private BigDecimal calculateFee(ParkingCost parkingCost) {
+		int hoursToPayFor;
+
+		/* calculating number of hours to pay for */
+		long differenceInMilis;
+		if (end != null) {
+			differenceInMilis = end.getTime() - start.getTime();
+		} else {
+			differenceInMilis = System.currentTimeMillis() - start.getTime();
+		}
+		hoursToPayFor = (int) Math.ceil(differenceInMilis / 3600000D);
+		/* end of calculating number of hours to pay for */
+
+		/* calculating the cost */
+		double fee = 0;
```

Comment:
dlaczego to nie jest liczone na BigDecimalach?

Reply:
Właściwie, było to dla uproszczenia (operacje na BigDecimalach są mniej wygodne), ale nie powinienem tak zrobić (powinno być na BigDecimalach).